### PR TITLE
backport: trivial 2024 10 23 pr3

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Batch of trivial backports

## What was done?
See commits

## How Has This Been Tested?
built locally; large combined merge passed tests locally

## Breaking Changes
Should be none

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Comment by UdjinM6** [comment]
guix fails
```
======================================================================
FAIL: test_MACHO (__main__.TestSymbolChecks)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/distsrc-base/distsrc-21.1.1-1147-g1613fce5d74a-arm64-apple-darwin/./contrib/devtools/test-symbol-check.py", line 99, in test_MACHO
    self.assertEqual(call_symbol_check(cc, source, executable, ['-lexpat', '-Wl,-platform_version','-Wl,macos', '-Wl,11.4', '-Wl,11.4']),
AssertionError: Tuples differ: (1, '[33 chars]D_LIBRARIES!\ntest1: failed DYNAMIC_LIBRARIES MIN_OS SDK LD64') != (1, '[33 chars]D_LIBRARIES!\ntest1: failed DYNAMIC_LIBRARIES MIN_OS SDK')

First differing element 1:
'libe[28 chars]ED_LIBRARIES!\ntest1: failed DYNAMIC_LIBRARIES MIN_OS SDK LD64'
'libe[28 chars]ED_LIBRARIES!\ntest1: failed DYNAMIC_LIBRARIES MIN_OS SDK'

  (1,
   'libexpat.1.dylib is not in ALLOWED_LIBRARIES!\n'
-  'test1: failed DYNAMIC_LIBRARIES MIN_OS SDK LD64')
?                                             -----

+  'test1: failed DYNAMIC_LIBRARIES MIN_OS SDK')

----------------------------------------------------------------------
```

I guess it's 28630

**Review by UdjinM6 — APPROVED** [review]
utACK c75a0d4c575edcc7288ec9423968921a6ad2ef58

File: depends/packages/qt.mk [hunk 1]
```diff
@@ -133,9 +133,6 @@ $(package)_config_opts_darwin += -no-feature-corewlan
 $(package)_config_opts_darwin += -no-freetype
 $(package)_config_opts_darwin += QMAKE_MACOSX_DEPLOYMENT_TARGET=$(OSX_MIN_VERSION)
 
-# Optimizing using > -O1 causes non-determinism when building across arches.
```
Comment (knst) [inline depends/packages/qt.mk:135]: 28778 - should we test it?

**Review by knst — COMMENTED** [review]
LGTM

beside not tested 28778

File: depends/packages/qt.mk (hunk 1 above)
Comment (PastaPastaPasta) [reply depends/packages/qt.mk:135]: CI on this branch gave
```
7e50766d89e933bf281fca317d0d0570595288535d1c0a9c7ab78d9ab62850e8  arm64-apple-darwin/dashcore-21.1.1-1146-gc75a0d4c575e-arm64-apple-darwin-debug.tar.gz
b268e492b334518c2ae7bc57a06db7c36e56bdaaae9806f1129f2fe33edb9f12  arm64-apple-darwin/dashcore-21.1.1-1146-gc75a0d4c575e-arm64-apple-darwin-unsigned.dmg
21c44d30c5664728f1a4cab8861c3a65aced61e5b838b33cfa5b1e8bff841570  arm64-apple-darwin/dashcore-21.1.1-1146-gc75a0d4c575e-arm64-apple-darwin-unsigned.tar.gz
0610721bc281e13581564fb1134c27a66d9e6a7ed2986f6c134fc2e5cd7fe495  arm64-apple-darwin/dashcore-21.1.1-1146-gc75a0d4c575e-arm64-apple-darwin.tar.gz
9f2d68a3584ab572c963570c3dffbdfe4ac7ce7f2ab477ef6330bc98f7ea5e24  dist-archive/dashcore-21.1.1-1146-gc75a0d4c575e.tar.gz
```

will provide hashes from my repo once complete

Comment (PastaPastaPasta) [reply depends/packages/qt.mk:135]: We should; I think let's merge it in and then we'll build guix here and on my repo, and compare. Revert if needed

**Pull request merged**